python/lsst/jointcal/utils.py

```diff
@@ -52,41 +58,72 @@ def compute_rms(self, data_refs, visit_list, reference):
 
         Return
         ------
-        geom.Angle
+        astropy.Quantity
             New relative RMS of the matched sources.
-        geom.Angle
+        astropy.Quantity
             New absolute RMS of matched sources.
         """
-        visits_per_dataRef = [dataRef.dataId['visit'] for dataRef in data_refs]
 
-        def compute(catalogs):
-            """Compute the relative and absolute RMS."""
+        # DECAM doesn't have "filter" in its registry, so we have to get filter names from VisitInfo.
+        self.filters = [ref.get('calexp').getInfo().getFilter().getName() for ref in data_refs]
+        visits_per_dataRef = [ref.dataId['visit'] for ref in data_refs]
+
+        def compute(catalogs, calibs):
+            """Compute the relative and absolute matches in distance and flux."""
             visit_catalogs = self._make_visit_catalogs(catalogs, visits_per_dataRef, visit_list)
             catalogs = list(visit_catalogs.values())
-            refcat = catalogs[0]  # use the first catalog as the relative reference catalog
-            relative = self._make_match_dict(refcat, catalogs[1:])
-            absolute = self._make_match_dict(reference, catalogs)
-            return visit_catalogs, relative, absolute
-
-        old_cats = [dataRef.get('src') for dataRef in data_refs]
-        visit_catalogs, self.old_relative, self.old_absolute = compute(old_cats)
-
-        # Update coordinates with the new wcs including distortion corrections.
+            # use the first catalog as the relative reference catalog
+            refcat = catalogs[0]
+            refcalib = calibs[0]
+            dist_rel, flux_rel, ref_flux_rel, source_rel = self._make_match_dict(refcat, catalogs[1:], calibs,
+                                                                                 refcalib=refcalib)
+            dist_abs, flux_abs, ref_flux_abs, source_abs = self._make_match_dict(reference, catalogs, calibs)
```

Comment:
I waffled several times about how to structure this data, but was not very happy about any of them, and would love to talk with you about ideas. shuffling things between various namedtuples doesn't seem like a huge improvement.